iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/ShortBackupRetentionCheck.java

```diff
@@ -46,7 +49,8 @@ public class ShortBackupRetentionCheck extends AbstractNewResourceCheck {
   protected void registerResourceConsumer() {
     register("aws_db_instance",
       resource -> {
-        if (resource.attribute("source_db_instance_identifier").isAbsent()) {
+        if (resource.attribute("source_db_instance_identifier").isAbsent()
+        && !resource.attribute("engine").is(expr -> TextUtils.getValue(expr).filter(ENGINES_EXCEPTION::contains).isPresent())) {
```

Comment:
Nitpick, move `expr -> TextUtils.getValue(expr).filter(ENGINES_EXCEPTION::contains).isPresent()` to a small function in this check to reduce the depth in this condition.